tweak(spells): Makes disintegrate a hand spell

sephirothseeker opened the pull request:
Заменяет выбор цели у заклинания гиба (ei'nath) с окошка на клик рукой по желаемой жертве. Выпилено проигрывание звука "bangindonk.ogg" при гибе жертвы с 5% шансом. 🥺 
Также перенёс это заклинание и заклинание mind_control, которое тоже работает через клик рукой, из папки targeted в соответствующую папку hand.

<details>
  <summary>Старое и неудобное окошко выбора цели</summary>
  
![spell-disintegrate-old](https://github.com/ChaoticOnyx/OnyxBay/assets/107106680/ba8609eb-7387-4678-9f1a-f152e0e1e4a0)

</details>
<details>
  <summary>Достаточно кликнуть рукой по жертве</summary>
  
![spell-disintegrate](https://github.com/ChaoticOnyx/OnyxBay/assets/107106680/b7c2b34f-b6c5-478d-83dd-1c68500fdfad)

</details>

<details>
<summary>Чейнджлог</summary>

```yml
🆑
tweak: Для использования заклинания Ei'Nath магу теперь достаточно кликнуть по своей жертве.
/🆑
```

</details>

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

sephirothseeker committed:
tweak(wizard spells): replaced disintegrate target input with wizard grab

Now instead of picking your target through input window, you just need to click your target with control hand.

also replaced hand_state for disintegrate and mind control with those from /tg/station.

sephirothseeker committed:
refact(spells): moved mind_control to hand

Moved mind_control from targeted spells to hand spells

NoneOfYourBsns commented on code/modules/spells/classes/battlemage.dm:
```diff
@@ -10,7 +10,7 @@
 		SPELL_DATA(/datum/spell/targeted/equip_item/shield,            1),
 		SPELL_DATA(/datum/spell/targeted/projectile/dumbfire/fireball, 1),
-		SPELL_DATA(/datum/spell/targeted/disintegrate,                 2),
+		SPELL_DATA(/datum/spell/hand/disintegrate,                 2),
```
Подровняй пробелами строку, а то выглядит странно.

NoneOfYourBsns commented on code/modules/spells/hand/disintegrate.dm:
```diff
@@ -0,0 +1,33 @@
+
+/datum/spell/hand/disintegrate/cast_hand(atom/A, mob/user)
+	var/mob/living/target = A
+	target.gib()
```
Где ньюлайн?

NoneOfYourBsns commented on code/modules/spells/hand/disintegrate.dm:
```diff
@@ -0,0 +1,33 @@
+		if(!M.put_in_active_hand(H))
+			qdel(H)
+			return
+	return 1
```
Почему не макрос `TRUE`?

Zert0X commented:
Стоп а зачем эй натху вообще какие-то клики, если там грабнут надо

sephirothseeker commented:
> Стоп а зачем эй натху вообще какие-то клики, если там грабнут надо

Скрины смотри, там наглядно показано, через какое извращение надо было пройти, и как теперь всё будет просто

Zert0X commented:
Таки не, я не про это, прост гиб кликом немного имба.

Гиб грабом и спелом - уже норм.

Не отрицаю, что то что было раньше - рак раковый.

sephirothseeker commented:
> Таки не, я не про это, прост гиб кликом немного имба.
> 
> Гиб грабом и спелом - уже норм.
> 
> Не отрицаю, что то что было раньше - рак раковый.

Рейндж старым остался в 1 тайл, тебе при касте даётся в руку кулачок и им нужно ударить по цели.

Zert0X commented:
> > Таки не, я не про это, прост гиб кликом немного имба.
> > 
> > Гиб грабом и спелом - уже норм.
> > 
> > Не отрицаю, что то что было раньше - рак раковый.
> 
> Рейндж старым остался в 1 тайл, тебе при касте даётся в руку кулачок и им нужно ударить по цели. 

Рендж и при грабе будет 1, просто при грабе это становиться хоть как-то контрибельно.

А и да спрайта кулака на кукле не вижу

sephirothseeker commented:
> > > Таки не, я не про это, прост гиб кликом немного имба.
> > > Гиб грабом и спелом - уже норм.
> > > Не отрицаю, что то что было раньше - рак раковый.
> > 
> > 
> > Рейндж старым остался в 1 тайл, тебе при касте даётся в руку кулачок и им нужно ударить по цели.
> 
> Рендж и при грабе будет 1, просто при грабе это становиться хоть как-то контрибельно.
> 
> А и да спрайта кулака на кукле не вижу

Сейчас стащу с тг

sephirothseeker commented:
> > > Таки не, я не про это, прост гиб кликом немного имба.
> > > Гиб грабом и спелом - уже норм.
> > > Не отрицаю, что то что было раньше - рак раковый.
> > 
> > 
> > Рейндж старым остался в 1 тайл, тебе при касте даётся в руку кулачок и им нужно ударить по цели.
> 
> Рендж и при грабе будет 1, просто при грабе это становиться хоть как-то контрибельно.
> 
> А и да спрайта кулака на кукле не вижу

Маг и так дерьмо паршивое, куда его ещё нерфить? И граб кодить надо, в билде нет такого, только через hand

Zert0X commented:
> > > > Таки не, я не про это, прост гиб кликом немного имба.
> > > > Гиб грабом и спелом - уже норм.
> > > > Не отрицаю, что то что было раньше - рак раковый.
> > > 
> > > 
> > > Рейндж старым остался в 1 тайл, тебе при касте даётся в руку кулачок и им нужно ударить по цели.
> > 
> > Рендж и при грабе будет 1, просто при грабе это становиться хоть как-то контрибельно.
> > 
> > А и да спрайта кулака на кукле не вижу
> 
> Маг и так дерьмо паршивое, куда его ещё нерфить? И граб кодить надо, в билде нет такого, только через hand

Лол всмысле нет, у тебя буквально граб есть, бери получай и проверяй, 3-5 строк кода

sephirothseeker commented:
> > > > > Таки не, я не про это, прост гиб кликом немного имба.
> > > > > Гиб грабом и спелом - уже норм.
> > > > > Не отрицаю, что то что было раньше - рак раковый.
> > > > 
> > > > 
> > > > Рейндж старым остался в 1 тайл, тебе при касте даётся в руку кулачок и им нужно ударить по цели.
> > > 
> > > 
> > > Рендж и при грабе будет 1, просто при грабе это становиться хоть как-то контрибельно.
> > > А и да спрайта кулака на кукле не вижу
> > 
> > 
> > Маг и так дерьмо паршивое, куда его ещё нерфить? И граб кодить надо, в билде нет такого, только через hand
> 
> Лол всмысле нет, у тебя буквально граб есть, бери получай и проверяй, 3-5 строк кода

Тогда сделай это

sephirothseeker committed:
refact(spells): minor changes

Removes unnecesary code, minor refact

sephirothseeker committed:
tweak(hand_item): adds item states for hand spells

Item states for hand spells. Sprties stolen from /tg/
I also added slim sprites but unfortunately they don't work

sephirothseeker commented:
Добавил айтем стейты с тг
И для слим версий тоже, но по какой-то причине они не работают

NoneOfYourBsns commented:
@sephirothseeker , у тебя еще в двух файлах пробелы съехали, недопустимо!

sephirothseeker committed:
refact(speels.dm): minor changes

kreeperHLC reviewed (dismissed):
Фикси слимов

sephirothseeker committed:
Merge branch 'dev-magic' of https://github.com/sephirothseeker/OnyxBay into dev-magic

sephirothseeker committed:
merging shit

sephirothseeker committed:
merging shit 2

NefrNerf commented on code/modules/spells/hand/disintegrate.dm:
```diff
@@ -1,34 +1,22 @@
-
+	hand_state = "disintegrate"
+	icon_state = "wiz_disint"
+	show_message = " puts his hand on target head, it's starting to glow brightly."
```
1) Отступ нужен?
2) his, а что если маг не мужчина а женщина или оно? Смени на что-то обезличенное по типу "Target`s head is starting to glow brightly". Только вот причем тут вообще свечение головы?

sephirothseeker replied:
1. Не нужен
2. Show_message не позволяет учитывать такие моменты

NefrNerf commented on code/modules/spells/hand/disintegrate.dm:
```diff
@@ -1,34 +1,22 @@
-			to_chat(user, "<span class='warning'>That was not so bright of you.</span>")
-			return
-		if(prob(5)) // Well why not?
-			playsound(user.loc, 'sound/misc/bangindonk.ogg', 50, 1)
```
А эта опция куда ушла?

sephirothseeker replied:
Какая опция?

NefrNerf commented on code/modules/spells/hand/disintegrate.dm:
```diff
@@ -1,34 +1,22 @@
+	compatible_targets = list(/mob/living)
 	cast_sound = 'sound/effects/squelch2.ogg'
 
-/datum/spell/targeted/disintegrate/cast(list/targets, mob/user)
```
Раньше метод в себя принимал список, а теперь один атом. Оставь метод с листом и сделай метод с одной целью, в методе с листом просто вызывай метод с одной целью, вообщем, вынеси общее.

sephirothseeker replied:
Зачем???

NefrNerf replied:
Потому что было до этого так. А что если дезинтегрейт нам нужен массовый? Если считаешь что в данном случае нужна только одна цель - не проблема, но не убирай метод, добавь новый

sephirothseeker replied:
Напиши и предложи к исправлению свой вариант изменений

NefrNerf replied:
Это ты взялся перепиливать дезинтигрейд и зачем-то спилил старый метод. Мое дело отревьювать и запросить правки

sephirothseeker replied:
Я не буду этого делать

NefrNerf replied:
¯\_(ツ)_/¯, Чр не сниму пока не сделаешь

No-Tips replied:
> Напиши и предложи к исправлению свой вариант изменений

Так просто же метод с листом, который бы перебирал этот лист и применял твой дизинтегрейт на каждый отдельный атом

NefrNerf reviewed (changes requested):
Кроме перечисленных изменений, у тебя еще майндконтроль почему-то хэнд стал

sephirothseeker commented:
> Кроме перечисленных изменений, у тебя еще майндконтроль почему-то хэнд стал

Читай описание пра

sephirothseeker committed:
tweak(disintegrate): desc

sephirothseeker committed:
Merge branch 'dev-magic' of https://github.com/sephirothseeker/OnyxBay into dev-magic

sephirothseeker committed:
merging shit 3
